DAL/user.DAL.js

```diff
@@ -0,0 +1,22 @@
+const User = require("./../modules/user.module");
+
+const addNewUser = async (user) => {
+  const newUser = User.create(user);
+  return newUser;
+};
+
+const getUserByEmail = async (req, res) => {
+  const foundUser = User.findOne({ email: req.body.email });
+  return foundUser;
+};
+
+const getUserById = async (req, res) => {
+  const foundUser = User.findOne({ id: req.body._id });
```

Comment:
`_id` is the mongo id? maybe consider using an internal id instead.

Reply:
this is mongo's id field key.
What else should I use?

Reply:
As explained in class
Either use this or generate one with external packages